File: app/src/main/java/pers/zhc/tools/test/viewtest/MainActivity.java
package pers.zhc.tools.test.viewtest;

import android.graphics.Color;
import android.os.Bundle;
import android.view.ViewGroup;
import pers.zhc.tools.BaseActivity;
import pers.zhc.tools.views.ColorShowRL;

public class MainActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ColorShowRL colorShowRL = new ColorShowRL(this);
        colorShowRL.setLayoutParams(new ViewGroup.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT));
        this.setContentView(colorShowRL);
        colorShowRL.setColor(Color.RED,"#FFFF0000");
    }
}

File: app/src/main/java/pers/zhc/tools/digest/MainActivity.kt
package pers.zhc.tools.digest

import android.os.Bundle
import androidx.core.widget.doAfterTextChanged
import pers.zhc.tools.BaseActivity
import pers.zhc.tools.databinding.DigestMainBinding
import pers.zhc.tools.jni.JNI
import pers.zhc.tools.utils.ClipboardUtils

class MainActivity: BaseActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val bindings = DigestMainBinding.inflate(layoutInflater)
        setContentView(bindings.root)

        val resultTv = bindings.resultTv

        bindings.textEt.editText.doAfterTextChanged {
            val hash = JNI.Digest.sha256(it.toString().toByteArray())
            resultTv.text = hash
        }

        resultTv.setOnClickListener {
            ClipboardUtils.putWithToast(this, resultTv.text.toString())
        }
    }
}
